megamek/src/megamek/common/DamageInfo.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (C) 2025 The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License (GPL),
+ * version 3 or (at your option) any later version,
+ * as published by the Free Software Foundation.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * A copy of the GPL should have been included with this project;
+ * if not, see <https://www.gnu.org/licenses/>.
+ *
+ * NOTICE: The MegaMek organization is a non-profit group of volunteers
+ * creating free software for the BattleTech community.
+ *
+ * MechWarrior, BattleMech, `Mech and AeroTech are registered trademarks
+ * of The Topps Company, Inc. All Rights Reserved.
+ *
+ * Catalyst Game Labs and the Catalyst Game Labs logo are trademarks of
+ * InMediaRes Productions, LLC.
+ *
+ * MechWarrior Copyright Microsoft Corporation. MegaMek was created under
+ * Microsoft's "Game Content Usage Rules"
+ * <https://www.xbox.com/en-US/developers/rules> and it is not endorsed by or
+ * affiliated with Microsoft.
+ */
+
+package megamek.common;
+
+import megamek.common.weapons.DamageType;
+
+public record DamageInfo(
+        Entity te,
+        HitData hit,
+        int damage,
+        boolean ammoExplosion,
+        DamageType damageType,
+        boolean damageIS,
+        boolean areaSatArty,
+        boolean throughFront,
+        boolean underWater,
+        boolean nukeS2S
+    ) {
+
+    public DamageInfo(
+          Entity te,
+          HitData hit,
+          int damage
+    ){
+        this(
+              te,
+              hit,
+              damage,
+              false,
+              DamageType.NONE,
+              false,
+              false
+        );
+    }
+
+    public DamageInfo(
+          Entity te,
+          HitData hit,
+          int damage,
+          boolean ammoExplosion
+    ){
+        this(
+              te,
+              hit,
+              damage,
+              ammoExplosion,
+              DamageType.NONE,
+              false,
+              false
+        );
+    }
```

Comment:
I know this completely personal preference, but I'd like to expose that my personal preference is to use static functions that build the class instead of having multiple constructors, this way there is only "one right way" to construct the class and all the others are effectively auxiliary functions to facilitate on that process.

Reply:
This is a lightweight dataclass to simplify the interface to TWDamageManager and reduce some of the multiple chained function calls in TWGameManager; adding any further functionality would just increase bug surface area and require further refactoring of TWGameManager.